File: TTS_ANGULAR_LEDUCHOA/BackEnd/airline_tickets_agency_management/src/main/java/com/backend/airline_tickets_agency_management/model/entity/flight_ticket/Ticket.java
package com.backend.airline_tickets_agency_management.model.entity.flight_ticket;

import com.backend.airline_tickets_agency_management.model.entity.user.User;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "ticket")
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ticket_id")
    private Long ticketId;
    private String ticketCode;
    private Boolean plusBaby;
    private Double plusBaggage;
    private String ticketType;
    private Double ticketTypePrice;
    private Double ticketPrice;
    private Double tax;
    private String chairName;
    @Column(columnDefinition = "date")
    private String bookingDate;
    private Integer checkInBaggage;
    private Integer carryOnBaggage;
    private String passengerType;
    private Double passengerTypePrice;
    private String passengerName;
    private String passengerGender;
    private String passengerPhone;
    private String passengerIdCard;
    private String passengerEmail;
    @ManyToOne
    @JoinColumn(name = "flight_id",referencedColumnName = "flight_id")
    private Flight flight;

    @ManyToOne
    @JoinColumn(name = "ticket_status_id",referencedColumnName ="ticket_status_id" )
    private TicketStatus ticketStatus;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private boolean flag = true;

}

File: TTS_ANGULAR_LEDUCHOA/FrontEnd/airline-tickets-agency-management/src/app/model/flight-ticket/ticket.ts

import {Flight} from './flight';

import {TicketStatus} from './ticket-status';
import {User} from '../user';


export interface Ticket {
  ticketId?: number;
  ticketCode?: string;
  plusBaby?: boolean;
  plusBaggage?: number;
  ticketType?: string;
  ticketTypePrice?: number;
  ticketPrice?: number;
  tax?: number;
  chairName?: string;
  bookingDate?: string;
  checkInBaggage?: number;
  carryOnBaggage?: number;
  passengerType?: string;
  passengerTypePrice?: number;
  passengerName?: string;
  passengerGender?: string;
  passengerPhone?: string;
  passengerIdCard?: string;
  passengerEmail?: string;
  flag?: boolean;
  flight?: Flight;
  ticketStatus?: any;
  user?: any;
}
